infra/worker/claimWorker.go

```diff
@@ -0,0 +1,346 @@
+			Status:         nil,
+		}).Error; err != nil {
+			return err
+		}
+		return nil
+	}); allTxErr != nil {
+		return 5, allTxErr
+	}
+	return 0, nil
+}
+
+func setupClaimWorkers() ([]*ClaimWorker, error) {
+	var workers []*ClaimWorker
+	var chains []models.Chain
+	if err := database.DB.Find(&chains).Error; err != nil {
+		return nil, err
+	}
+	for _, chain := range chains {
+
+		if err := blockchain.AddChainClient(chain); err != nil {
+			return nil, err
+		}
+
+		secret, err := models.GetSecret(chain.Id)
+		if err != nil {
+			logrus.WithFields(logrus.Fields{"chain": chain.Id, "name": chain.Name}).Error("secret not found")
+			return nil, err
+		}
+		// prepare claiming cursor
+		var claimCursor models.ClaimCursor
+		err = database.DB.Where("target_chain=?", chain.Id).Take(&claimCursor).Error
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			if err = database.DB.Create(&models.ClaimCursor{TargetChain: chain.Id, CrossInfoId: 0}).Error; err != nil {
+				logrus.WithFields(logrus.Fields{"chainId": chain.Id}).
+					Error("create claim cursor error")
+				return nil, err
+			}
+		} else if err != nil {
+			logrus.WithFields(logrus.Fields{"chainId": chain.Id}).
+				Error("take claim cursor error")
+			return nil, err
+		}
+
+		// prepare private key
+		privateKey, address, err := blockchain.CreateKeyPair(secret.Private[2:])
+		if err != nil {
+			logrus.WithError(err).WithFields(logrus.Fields{
+				"chainId": chain.Id,
+			}).Error("create key pair error")
+			return nil, err
+		}
+		evmHandler := blockchain.GetEvmHandler(chain.Id)
+
+		if secret.Address == "" {
+			if err := database.DB.Model(&secret).Where("id=?", secret.Id).
+				Update("address", address.Hex()).Error; err != nil {
+				logrus.WithError(err).WithFields(logrus.Fields{
+					"chainId": chain.Id, "secretId": secret.Id,
+				}).Error("update address for secret fail")
+				return nil, err
+			}
+			balance, err := evmHandler.Client.BalanceAt(context.Background(), *address, nil)
+			if err != nil {
+				logrus.WithError(err).WithFields(logrus.Fields{
+					"chainId": chain.Id, "address": *address,
+				}).Error("get balance fail")
+				return nil, err
+			}
+			if (*balance).Cmp(big.NewInt(0)) == 0 {
+				logrus.WithError(err).WithFields(logrus.Fields{
+					"chainId": chain.Id, "address": *address,
+				}).Fatal("balance is zero")
+			}
+		}
+
+		workers = append(workers, &ClaimWorker{
+			ChainId:       chain.Id,
+			keyPair:       privateKey,
+			address:       address,
+			evmHandler:    evmHandler,
+			DelayForError: 60 * 10, // delay 10 minutes
+		})
+	}
+	return workers, nil
+}
+
+func (worker ClaimWorker) Claim(crossInfo models.CrossInfo) (string, *uint64, error) {
+	var items []models.CrossItem
+	if err := database.DB.Where("cross_info_id=?", crossInfo.Id).Find(&items).Error; err != nil {
+		logrus.WithError(err).WithFields(logrus.Fields{
+			"crossId": crossInfo.Id, "chain": crossInfo.TargetChain,
+		}).Error("cross items not found")
```

Comment:
error message `cross item not found` is not accurate

Reply:
Changed to  "querying CrossItems in DB fail"